### PR TITLE
ShaderNode: Improve syntax (Fluent interface)

### PR DESCRIPTION
**Pull request by sunag**
## Description

This PR adds [Fluent interface](https://en.wikipedia.org/wiki/Fluent_interface) for `ShaderNode`
Now is possible to use [method chaining](https://en.wikipedia.org/wiki/Method_chaining) between all `NodeElements`.

I think this can simplify the code reading, besides being common among Three.js users.

/ping @mrdoob 

## Examples

One line of `DFGApprox`:
```js
// previous
const r = add( mul( roughness, c0 ), c1 );

// using method chaining
const r = roughness.mul( c0 ).add( c1 );
```

Simple negate
```js
// previous
reflectVec = negate( positionViewDirection );

// using method chaining
reflectVec = positionViewDirection.negate();
```

Others examples
```js
//  reflectVec  = transformDirection( reflectVec, cameraViewMatrix );
reflectVec = reflectVec.transformDirection( cameraViewMatrix );

// invert( radianceTextureUVNode.y );
radianceTextureUVNode.y.invert();
```

*This contribution is funded by [Google](https://google.com)*

### CODE REVIEW

File: examples/jsm/nodes/functions/BSDF/DFGApprox.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const DFGApprox = new ShaderNode( ( inputs ) => {
 
 	const c1 = vec4( 1, 0.0425, 1.04, - 0.04 );
 
-	const r = add( mul( roughness, c0 ), c1 );
+	const r = roughness.mul( c0 ).add( c1 );
 
 	const a004 = add( mul( min( mul( r.x, r.x ), exp2( mul( - 9.28, dotNV ) ) ), r.x ), r.y );
```
Comment (LeviPesin) [inline examples/jsm/nodes/functions/BSDF/DFGApprox.js:19]: Can't we change `mul( r.x, r.x )` to `r.x.mul( r.x )` and make similar changes in other cases?

**Comment by sunag** [comment]
How it works:

1. Checks that no variable with the element name has been created on the target Node.
2. Returns a closure function, where the function's first argument is the node itself applied to the new element.
```js
// api
myNode.add( 100 );

// shader node (e.g.)
return ( ...params ) => add( myNode, ...params );
```

File: examples/jsm/nodes/functions/BSDF/F_Schlick.js [hunk 2]
```diff
@@ -9,9 +9,9 @@ const F_Schlick = new ShaderNode( ( inputs ) => {
 
 	// Optimized variant (presented by Epic at SIGGRAPH '13)
 	// https://cdn2.unrealengine.com/Resources/files/2013SiggraphPresentationsNotes-26915738.pdf
-	const fresnel = exp2( mul( sub( mul( - 5.55473, dotVH ), 6.98316 ), dotVH ) );
+	const fresnel = exp2( mul( - 5.55473, dotVH ).sub( 6.98316 ).mul( dotVH ) );
```
Comment (LeviPesin) [inline examples/jsm/nodes/functions/BSDF/F_Schlick.js:12]: Why not `dotVH.mul( - 5.55473 ).sub( 6.98316 ).mul( dotVH ).exp2()`?

File: examples/webgpu_compute.html [hunk 3]
```diff
@@ -97,7 +97,7 @@
 					stack.assign( position, max( negate( limit ), min( limit, position ) ) );
 
 					const pointerSize = 0.1;
-					const distanceFromPointer = length( sub( pointer, position ) );
+					const distanceFromPointer = pointer.sub( position ).length();
 
 					stack.assign( particle, cond( lessThanEqual( distanceFromPointer, pointerSize ), vec3(), position ) );
```
Comment (LeviPesin) [inline examples/webgpu_compute.html:102]: Maybe also move to the new syntax on this line? `distanceFromPointer.lessThanEqual( pointerSize )` looks much nicer.

**Pull request merged**